smbc/file.c

```diff
@@ -152,7 +154,9 @@ File_read (File *self, PyObject *args)
     {
       fn_fstat = smbc_getFunctionFstat (ctx->context);
       (*fn_fstat) (ctx->context, self->file, &st);
-      size = st.st_size;
+      fn_lseek = smbc_getFunctionLseek (ctx->context);
+      current = (*fn_lseek) (ctx->context, self->file, 0, 1);
```

Comment:
As no real seek is involved, I guess this should never fail, but a check to see if it is `-1` could be added (what would happen then?).